urllib3/connection.py

```diff
@@ -205,11 +206,17 @@ def set_cert(self, key_file=None, cert_file=None,
         self.key_file = key_file
         self.cert_file = cert_file
         self.cert_reqs = cert_reqs
-        self.ca_certs = ca_certs
-        self.ca_cert_dir = ca_cert_dir
         self.assert_hostname = assert_hostname
         self.assert_fingerprint = assert_fingerprint
 
+        if ca_certs:
+            ca_certs = os.path.expanduser(ca_certs)
+        if ca_cert_dir:
+            ca_cert_dir = os.path.expanduser(ca_cert_dir)
```

Comment:
Hm this line needs test coverage now. :) Could rewrite it as one line if we'd prefer.

```diff
- self.ca_certs = ca_certs
- self.ca_cert_dir = ca_cert_dir
+self.ca_certs = ca_certs and os.path.expanduser(ca_certs)
+self.ca_cert_dir = ca_cert_dir and os.path.expanduser(ca_cert_dir)
```